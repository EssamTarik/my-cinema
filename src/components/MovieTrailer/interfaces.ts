import { IMovieTrailer } from "../../store/movieTrailer/models";

export interface IProps {
  trailer: IMovieTrailer
}