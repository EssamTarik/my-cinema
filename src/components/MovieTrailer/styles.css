.movie-trailer {
  width: 75vw;
}