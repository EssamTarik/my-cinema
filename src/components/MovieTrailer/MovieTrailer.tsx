import React from 'react';
import Youtube from 'react-youtube'
import { IProps } from './interfaces';
import './styles.css';

const MovieTrailer = ({ trailer }: IProps) => {
  const { source } = trailer;
  return (
    <div className='movie-trailer'>
      <Youtube opts={{ width: '100%' }} videoId={source} />
    </div>
  )
}

export default MovieTrailer;