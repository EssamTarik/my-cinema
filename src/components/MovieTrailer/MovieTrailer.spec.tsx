jest.mock('react-youtube', () => () => <div />)
jest.mock('react-intl', () => ({ FormattedMessage: () => <div /> }));
import React from "react";
import { render, cleanup } from "@testing-library/react";

const trailerMock = jest.requireMock('trailer');

import MovieTrailer from "./MovieTrailer";

describe('MovieTrailer', () => {
  afterEach(cleanup);
  it("renders", () => {
    const { asFragment } = render(
      <MovieTrailer trailer={trailerMock} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
})

