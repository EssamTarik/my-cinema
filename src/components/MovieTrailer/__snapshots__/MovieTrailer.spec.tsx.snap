// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTrailer renders 1`] = `
<DocumentFragment>
  <div
    class="movie-trailer"
  >
    <div />
  </div>
</DocumentFragment>
`;
